course_content/solutions/numpy_exercise_7.ipynb

```diff
@@ -0,0 +1,112 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise 7\n",
+    "\n",
+    "Sometimes an operation will produce a result, but not the one you wanted.\n",
+    "\n",
+    "For example, suppose we have :\n",
+    "```python\n",
+    "A = np.array([[1, 2, 3],\n",
+    "              [4, 5, 6],\n",
+    "              [7, 8, 9]])\n",
+    "```\n",
+    "\n",
+    "and we wish to add 0, 100 and 400 ***to the rows***.\n",
+    "That is,\n",
+    "\n",
+    "```python\n",
+    "B = np.array([0, 100, 400])\n",
+    "# and...\n",
+    "desired_result = np.array([[  1,   2,   3],\n",
+    "                           [104, 105, 406],\n",
+    "                           [407, 408, 409]])\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Questions:\n",
+    "\n",
+    " 1. What will be the result of adding A to B directly?\n",
+    " 1. How can you perform the correct calculation on `A` and `B` to get `desired_result`?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "A = np.array([[1, 2, 3],\n",
+    "              [4, 5, 6],\n",
+    "              [7, 8, 9]])\n",
+    "\n",
+    "B = np.array([0, 100, 400])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "A + B"
```

Comment:
It is not clear which cells are answering question 1 and which ones question 2. It would be worth consolidating the cells answering question 1 into one cell, and the same for question 2.